src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs

```diff
@@ -150,16 +152,26 @@ public static VectorToImageConvertingEstimator ConvertToImage(this TransformsCat
         /// <param name="width">The width of the output images.</param>
         /// <param name="outputColumnName"> Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName"> Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
-        /// <param name="colors"> Specifies which <see cref="ImagePixelExtractingEstimator.ColorBits"/> are in the input pixel vectors. The order of colors is: Alpha, Red, Green Blue.</param>
-        /// <param name="interleave">Whether the pixels are interleaved, meaning whether they are in `ARGB ARGB` order, or separated in the planar form, where the colors are specified one by one
-        /// alpha, red, green, blue for all the pixels of the image. </param>
+        /// <param name="colors"> Specifies which <see cref="ImagePixelExtractingEstimator.ColorBits"/> are in the input pixel vectors. The order of colors specified in <paramref name="order"/></param>
+        /// <param name="order">In which order extract colors presented in array.</param>
+        /// <param name="interleave">Whether the pixels are interleaved, meaning whether they are in <paramref name="order"/> order, or separated in the planar form, where the colors are specified one by one
+        /// for all the pixels of the image. </param>
         /// <param name="scale">The values are scaled by this value before being converted to pixels.</param>
         /// <param name="offset">The offset is subtracted (before scaling) before converting the values to pixels.</param>
+        /// <param name="defaultAlpha">Default value for alpha color, would be overriden if <paramref name="colors"/> contains <see cref="ImagePixelExtractingEstimator.ColorBits.Alpha"/>.</param>
```

Comment:
I think your code now has the opposite logic.

Would prefer to be consistent with the above extension here too. If you could also reorder them so that the first one applied comes first that would be great.

Reply:
What do you mean?

---
In reply to: [259550143](https://github.com/dotnet/machinelearning/pull/2602#discussion_r259550143) [](ancestors = 259550143)

Reply:
Are you talking about scale offset or defaultAlpha?

---
In reply to: [259550340](https://github.com/dotnet/machinelearning/pull/2602#discussion_r259550340) [](ancestors = 259550340,259550143)

Reply:
Sorry I think codeflow did not pick up the right location for this. I meant to comment on `scale` and `offset`. 

1. I think the arguments should be presented in the same order in which they are applied.
2. I think the order and the comment should match the above extension for ImageLoader
3. I think the logic that you have in your code here does not match the comment: https://github.com/dotnet/machinelearning/blob/cee009e90767b4bd991813809e9425dbc11d17f5/src/Microsoft.ML.ImageAnalytics/VectorToImageTransform.cs#L415-L419
It seems that you are first scaling and the applying the offset.

Reply:
So I can do this: P-pixel x -VectorValue
p= x*scale1-offset1
x =p*scale2-offset2

this way I need to solve equation in order to pick up scale1 and offset1 if I know scale2 and offset2.
or I can have 

x=(p-offset1)*scale1
p = x*scale2 - offset2
and relationship between offset1 = -offset2 and scale1 = 1/scale2


So in one transform i apply scale first, in another i apply offset.
I can make them same order in their parameters, but since I'm specifying which applied first and which second, it looks weird then you put one which applied second to first in argument list.



---
In reply to: [259551787](https://github.com/dotnet/machinelearning/pull/2602#discussion_r259551787) [](ancestors = 259551787)

Reply:
Thank you for explaining the logic it makes sense like this!


---
In reply to: [259553517](https://github.com/dotnet/machinelearning/pull/2602#discussion_r259553517) [](ancestors = 259553517,259551787)